concord-server/server/src/main/java/com/rabidgremlin/concord/dao/LabelsDao.java

```diff
@@ -21,4 +23,18 @@
 	@SqlQuery("SELECT * FROM labels ORDER BY label")
     @RegisterBeanMapper(Label.class)
     List<Label> getLabels();
+
+	@Transaction
+	default void updateLabels(List<Label> labels) {
+		deleteAllLabels();

```

Comment:
Maybe pull this out to the uploadCsv method, as deleting all labels isn't strictly update functionality.

Reply:
Having it in this class ensures that all the database calls use the same handler - which is needed to ensure they occur in a transaction. 

If we move into the uploadCsv method, we will have to do more lower level handling of the database connection